blackbird/src/main/java/com/fasterxml/jackson/module/blackbird/CrossLoaderAccess.java

```diff
@@ -109,24 +109,29 @@ private static Class<?> accessClassIn(MethodHandles.Lookup lookup) throws IOExce
         try {
             return Class.forName(pkg.getName() + "." + CLASS_NAME, true, lookup.lookupClass().getClassLoader());
         } catch (ClassNotFoundException ign) { }
-        String fqcn = pkg.getName()
-                .replace('.', '/')
-            + "/" + CLASS_NAME;
-        ByteArrayOutputStream classBytes = new ByteArrayOutputStream(HEADER.length + FOOTER.length + fqcn.length() + 16);
-        DataOutputStream dataOut = new DataOutputStream(classBytes);
-        for (int b : HEADER) {
-            dataOut.writeByte(b);
-        }
-        dataOut.writeUTF(fqcn);
-        for (int b : FOOTER) {
-            dataOut.writeByte(b);
-        }
-        try {
-            return (Class<?>) DEFINE_CLASS.invokeExact(lookup, classBytes.toByteArray());
-        } catch (RuntimeException | Error | IOException | ReflectiveOperationException e) {
-            throw e;
-        } catch (Throwable e) {
-            throw new RuntimeException(e);
+        synchronized(CrossLoaderAccess.class) {
+            try {
+                return Class.forName(pkg.getName() + "." + CLASS_NAME, true, lookup.lookupClass().getClassLoader());
+            } catch (ClassNotFoundException ign) { }
```

Comment:
"try again" logic is attempting to do double checked locking -- I think that's ok in this case but could always use more eyes on those attempts.  Could also remove the unsynchronized reflection and just do everything in a sync block?

Reply:
I think it's nice to avoid unnecessary synchronization in the common path - where the class is already loaded.
It's a little unfortunate that this `Class.forName` expression is repeated, but factoring it out might not be much clearer.

What about computing the className and fetching classLoader once, stash them in local vars - then at least we don't do the string concatenation twice unnecessarily.